modules/ROOT/pages/connector-migration.adoc

```diff
@@ -0,0 +1,139 @@
+
+Some parameters have to be passed to the maven archetype to create the project (groupid, artifactId, version, language...). Make sure to use *Java* as the language, so the existing implementation code will be reusable.
+
+[TIP]
+====
+To ease the migration, it is recommended to use the same definition id and definition version of the current connector. +
+To do so, the maven properties `<connector-definition-id>` and `<connector-definition-version>` have to be edited in the generated pom.xml of the project with the correct values. By default, it takes the project artifact id as definition id and the project version as definition version.
+====
+
+== 2 - Declare the connector dependencies
+
+A connector often depend on external Java libraries. +
+Previously, they were declared in the implementation and sometimes in the definition. A jar file had to be imported in the Bonita Studio, and then it was possible for a connector to depend on this jar. +
+Now, connector projects are Maven projects, dependencies are handled by Maven. It means that all the connector dependencies have to be declared in the file `pom.xml` (at the root of the project) as maven dependencies. https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html[Learn more about the Maven dependency mechanism].
+
+For each declared dependency in you implementation or in your definition, *the corresponding Maven dependency has to be found and declared in the pom.xml*. +
+For example, if a connector depends on the Jar `commons-lang3-3.12.0.jar`, then the corresponding Maven dependency is the dependency `Apache Commons Lang` in version `3.12.0`. So, the following Maven dependency has to be declared in the `pom.xml`: 
+
+[source, xml]
+----
+<dependency>
+    <groupId>org.apache.commons</groupId>
+    <artifactId>commons-lang3</artifactId>
+    <version>3.12.0</version>
+</dependency>
+----
+
+Finding the corresponding maven dependency of a Jar file can be hard, even impossible if the Jar has never been deployed on any Maven repository.
+
+The first thing to do is to simply copy past the Jar name in a web search engine, and hopefully the corresponding maven dependency will show up. You can also try to look for it on famous public maven repositories, like https://mvnrepository.com/[Maven central].
+
+In case it doesn't work, you can try to explore the Jar file (using an archive manager) to see which class it contains. +
+For example, if you explore the jar `commons-lang3-3.12.0.jar`, you will find a class named `StringUtils`. If you type _StringUtils maven_ in a web search engine you have a good chance to find the corresponding maven dependency.
+
+If the corresponding Maven dependency cannot be found publicly, it can mean that it is a private dependency of your organization. In this case, ask your IT team a way to retrieve this dependency through Maven, and to consider to deploy it on a maven repository (private or public) if it's not the already case.
```

Comment:
Not sure if it is relevant here

Reply:
I disagree, I'm pretty sure it can often happen.